### PR TITLE
Update plotting script

### PR DESCRIPTION
**Pull request by mmasciov**
In detail:
- allow for multiple years;
- allow for only distributions inclusive in mll;
- allow for only specific selections;
- only plot two mass points/signal for distributions other than m(ll);
- adjust plotting style accordingly.

### CODE REVIEW

File: python/plotUtils.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        thisPoint = g_ratio.GetN()
+        g_ratio.SetPoint(thisPoint, x, r)
+        g_ratio.SetPointError(thisPoint, xerr, xerr, rerrdown, rerrup)
+
+def GetEfficRatioGraph(hnum, hden, g_ratio):
+    level = 0.6827
+    for i in range(1, hnum.GetNbinsX()+1):
+        x = hnum.GetBinCenter(i)
+        xerr = hnum.GetBinWidth(i) / 2.0
+        num = int(hnum.GetBinContent(i))
```
Comment (VourMa) [inline python/plotUtils.py]: Why convert this into a integer? Don't we want float operations?

File: python/stack_plots.py [hunk 2]
```diff
@@ -29,6 +31,19 @@
 if len(args.signalMass)==0: 
     args.signalMass = [200,400,700,1000,1500,2000]
 
+massToExclude=[]
+if len(args.signalMass)>2:
+    for i,m in enumerate(args.signalMass):
+        if i==0 or i==len(args.signalMass)-1:
```
Comment (VourMa) [inline python/stack_plots.py]: In the general case that we plot all mass points for the mll plot, should we also have an intermediate mass point, say 700 or 1000 GeV, for the rest of the plots as well (apart from the 200 and 2000 GeV ones that I understand are plotted by default)?

File: python/stack_plots.py [hunk 3]
```diff
@@ -131,59 +162,96 @@
...
-        if sample=="Y3" or sample=="DY3" or sample=="DYp3" or sample=="B3mL2":
-            for mass in args.signalMass: 
-                 sampleDict[sample+"_M"+str(mass)]=ROOT.TFile(args.inDir+"output_"+sample+"_M"+str(mass)+"_2018.root")
-        elif sample=="ZToMuMu":
-            for m1,m2 in zip(["50","120","200","400","800","1400","2300","3500","4500","6000"],["120","200","400","800","1400","2300","3500","4500","6000","Inf"]): 
-                sampleDict[sample+"_"+m1+"_"+m2]=ROOT.TFile(args.inDir+"output_ZToMuMu_"+m1+"_"+m2+"_2018.root")
-        else:
-            sampleDict[sample]=ROOT.TFile(args.inDir+"output_"+sample+"_2018.root")
-
+    if year!="all":
```
Comment (VourMa) [inline python/stack_plots.py:171]: I understand that the following loops over samples are exactly the same, except for the loop over years. If that's the case, maybe they can be simplified by writing:

`if year!="all":`
__`years=[year]`
`else:`
__`years=["2016nonAPV","2016APV","2017","2018"]`

and then write just the loop from line 190?

Or is there a difference between the loop in lines 173-187 and the loop in lines 192-205?

File: python/stack_plots.py [hunk 4]
```diff
@@ -368,74 +465,158 @@ def draw_plot(sampleDict, plotname, logY=True, logX=False, plotData=False, doRat
...
-    stack.GetYaxis().SetLabelSize(0.03)
-    stack.GetYaxis().SetMaxDigits(3)
+        h_axis.GetYaxis().SetTitle("A.U.")
+    else:
+        h_axis.GetYaxis().SetTitle(totalSM.GetYaxis().GetTitle())
+    h_axis.GetYaxis().SetLabelSize(0.03)
+    h_axis.GetYaxis().SetMaxDigits(3)
+    h_axis.Draw("")
+    stack.Draw("HIST,SAME")
+    g_unc.Draw("SAME,2")
```
Comment (VourMa) [inline python/stack_plots.py:576]: What does the option `2` do?

File: python/plotUtils.py (hunk 1 above)
Comment (mmasciov) [reply python/plotUtils.py]: I believe this was due to the fact that number of passed and total events are normally integer in data.
However, floats will work and should not make a difference. So, will remove the conversion.

File: python/stack_plots.py (hunk 2 above)
Comment (mmasciov) [reply python/stack_plots.py]: Yes, I wasn't sure what the best choice could be, so I only picked the first and the last.
ButI can pick for instance every odd or even mass point, to make this semi-automatized?

File: python/stack_plots.py (hunk 4 above)
Comment (mmasciov) [reply python/stack_plots.py:576]: Option "2" allows to draw errors as rectangles (here rendered as dashes/grey area).

File: python/stack_plots.py (hunk 2 above)
Comment (VourMa) [reply python/stack_plots.py]: Yes, that works, thanks.

Comment (mmasciov) [reply python/stack_plots.py]: Ok, selection of even mass points is implemented.

File: python/stack_plots.py (hunk 3 above)
Comment (mmasciov) [reply python/stack_plots.py:171]: Right, thanks. Done.

**Pull request merged**